dataservice/api/common/error_handler.py

```diff
@@ -0,0 +1,62 @@
+"""
+API level error handling/message formatting
+
+Includes helpers to parse the generic database integrity error
```

Comment:
There doesn't seem to be a better way to distinguish specific types of integrity errors. If someone else knows of anything better please let me know.